strax/storage/mongo.py

```diff
@@ -40,15 +38,16 @@ def _read_chunk(self, backend_key, chunk_info, dtype, compressor):
         # Query for the chunk and project the chunk info
         doc = self.db[self.col_name].find_one(
             {**query, chunk_name: {"$exists": True}},
-            {f"{chunk_name}.$": 1})
-        doc = doc.get(chunk_name, None)
+            {f"{chunk_name}": 1})
 
-        if doc is None:
+        # Unpack info about this chunk from the query. Return empty if not available.
+        if doc is None or chunk_name not in doc:
             return np.array([], dtype=dtype)
-
+        else:
+            chunk_doc = doc[chunk_name]
```

Comment:
The c++ part of my brain (which is most of it right now) is complaining that `chunk_doc` goes out of scope after this line and so shouldn't exist when it's used later on, but it works. Though, the most pedantic observers could argue that because the `if` statement includes a return, the `if-else` construct doesn't actually do anything beyond `if` alone.

Reply:
The 'else' makes it more explicit this is arguably preferable as otherwise someone might overlook that the if statement checks that the ```doc[chunk_name]``` will not fail. On the internet you can find many pros/cons for these kind of else statements. See e.g. https://dmerej.info/blog/post/else-after-return-yea-or-nay/. 

I don't think it matters so if you think it's better I'm fine with it.